backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java

```diff
@@ -292,7 +293,10 @@ public IGitMacheteRepositorySnapshot discoverLayoutAndCreateSnapshot() throws Gi
 
     IGitMacheteRepositorySnapshot createGitMacheteRepository(IBranchLayout branchLayout) throws GitMacheteException {
       var rootBranchTries = branchLayout.getRootEntries().map(entry -> Try.of(() -> createGitMacheteRootBranch(entry)));
-      var rootBranches = Try.sequence(rootBranchTries).getOrElseThrow(GitMacheteException::getOrWrap).toList();
+      var rootBranchesCreationResults = Try.sequence(rootBranchTries).getOrElseThrow(GitMacheteException::getOrWrap);
+      var rootBranches = rootBranchesCreationResults.map(creationResult -> creationResult.getCreatedRootBranch()).toList();
+      var notCreatedBranches = rootBranchesCreationResults.flatMap(creationResult -> creationResult.getNotCreatedBranchNames())
+          .toList();
```

Comment:
Nit: `toList()` can be once on `rootBranchesCreationResults` instead

Reply:
Nice!